### PR TITLE
Run `live` through Tor

### PR DESCRIPTION
**Pull request by sectore**
Reference:
   - `socat` https://linux.die.net/man/1/socat
   - `socator` https://github.com/beacloudgenius/socator
   - BTCPay Server -> Reverse proxy to Tor https://docs.btcpayserver.org/Deployment/ReverseProxyToTor/#do-all-this-in-a-docker-container

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -32,11 +33,18 @@ run-services:
   # if someone has a better solution, be my guest
   sed 's/GATEWAY_IPV4/'` ip addr show eth0 | grep inet | awk '{print $2}' | cut -d/ -f1 | head -n1`'/g' \
     docker/docker-compose.monkey-patch-linux.yml | tee tmp/docker-compose.linux.yml
-  cd docker && docker compose -f docker-compose.yml -f ../tmp/docker-compose.linux.yml up
```
Comment (FelixWeis) [inline justfile:35]: why was this removed?

File: src/backend/bitcoin_core.rs [hunk 2]
```diff
@@ -24,7 +23,9 @@ pub struct BitcoinCore {
 //#[async_trait]
 impl BitcoinCore {
   pub fn new() -> Self {
-    let core_url = var("CORE_URL").unwrap_or("127.0.0.1:8332".to_owned());
+    let core_address = var("CORE_ADDRESS").unwrap_or("127.0.0.1".to_owned());
+    let core_port = var("CORE_PORT").unwrap_or("8332".to_owned());
+    let core_url = format!("{}:{}", core_address, core_port);
```
Comment (FelixWeis) [inline src/backend/bitcoin_core.rs:28]: good idea to split it

File: src/backend/bitcoin_core.rs [hunk 3]
```diff
@@ -102,7 +103,7 @@ impl BitcoinCore {
     // let data = hex::decode(&hex)?;
     // let transaction: Transaction = bitcoin::consensus::deserialize(&data)?;
     let transaction = client.get_raw_transaction(&txid, None)?;
-    //dbg!(&transaction);
+    // dbg!(&transaction);
```
Comment (FelixWeis) [inline src/backend/bitcoin_core.rs:106]: lets remove all commented code in this fn

File: justfile [hunk 4]
```diff
@@ -32,11 +33,18 @@ run-services:
...
   sed 's/GATEWAY_IPV4/'` ip addr show eth0 | grep inet | awk '{print $2}' | cut -d/ -f1 | head -n1`'/g' \
     docker/docker-compose.monkey-patch-linux.yml | tee tmp/docker-compose.linux.yml
-  cd docker && docker compose -f docker-compose.yml -f ../tmp/docker-compose.linux.yml up 
 
 clean-services:
   cd docker && docker compose down -v
 
+tor:
+  docker run --rm -ti \
+    -p 5000:5000 \
```
Comment (FelixWeis) [inline justfile:42]: socator port could be set via .env

**Review by FelixWeis — CHANGES_REQUESTED** [review]

File: justfile (hunk 1 above)
Comment (sectore) [reply justfile:35]: Removed by an accident - will revert it.

File: justfile (hunk 4 above)
Comment (sectore) [reply justfile:42]: > socator port could be set via .env

We can't set it, `socator` has `5000` (see hardcoded https://github.com/beacloudgenius/socator/blob/main/tor-socat.sh#L25)

Comment (FelixWeis) [reply justfile:43]: only for the docker internal port is hardcoded, I believe. Thats fine. The forwarded port on the host should be able to change because some environments might already use something on localhost:5000

Comment (sectore) [reply justfile]: Ok, updated in e1c0605

**Review by FelixWeis — APPROVED** [review]

**Comment by Emzy** [comment]
ACK socat

Setup:
Bitcoin RCP host: Raspberry PI4 4GB with Bitcoin Core (RaspiBlitz)
Live: Raspberry PI4 8GB

socator version is not working for me on my Raspberry PI4 8GB.
The error I getting is:
```
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
```

It is very slow. It's 10 Mempool TX every 44 seconds for me. It would take 13h to parse a 10000 TX mempool.
Using ssh tunnel between my RaspiBlitz and the PI running Live (LAN connection) is much faster (5min):
```
Opening SSH Tunnel:
admin@raspiBlitz:~$ ssh -R 8332:127.0.0.1:8332 user@live-pi
```

I think TOR for RPC is not very practical. It can't keep up with the Mempool for me.
I'm not surprised that RPC over TOR is very slow, TOR has a very long round trip time.

**Comment by FelixWeis** [comment]
thanks for testing this. we're going to speed up by a bit by using mempool REST in paralell but TOR will stay much slower than loopback
